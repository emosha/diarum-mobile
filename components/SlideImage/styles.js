import { StyleSheet, Dimensions } from 'react-native';

const dimensions = Dimensions.get('window');
const imageWidth = dimensions.width;

const styles = StyleSheet.create({
  imageStyle: {
    height: '100%',
    width: imageWidth,
  },
});

export default styles;
