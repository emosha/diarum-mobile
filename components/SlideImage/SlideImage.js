import React from 'react';
import PropTypes from 'prop-types';
import { Image } from 'react-native';
import styles from './styles';

export default class SlideImage extends React.Component {
  static propTypes = {
    imageSrc: PropTypes.any.isRequired,
  }

  render() {
    return (
      <Image
        source={this.props.imageSrc}
        style={styles.imageStyle}
      />
    );
  }
}
