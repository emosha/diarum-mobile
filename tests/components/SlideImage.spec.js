import React from 'react';
import SlideImage from '../../components/SlideImage/SlideImage';

test('renders correctly', () => {
  const tree = renderer.create(<SlideImage />).toJSON();
  expect(tree).toMatchSnapshot();
});
