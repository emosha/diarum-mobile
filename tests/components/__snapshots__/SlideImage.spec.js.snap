// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Image
  source={
    Object {
      "uri": undefined,
    }
  }
  style={
    Object {
      "height": "100%",
      "width": 750,
    }
  }
/>
`;
